components/script/dom/bindings/structuredclone.rs

```diff
@@ -67,16 +67,62 @@ unsafe fn read_length(r: *mut JSStructuredCloneReader)
   return length as usize;
 }
 
+struct StructuredCloneWriter {
+    w: *mut JSStructuredCloneWriter,
+}
+
+impl StructuredCloneWriter {
+    unsafe fn write_slice(&self, v: &[u8]) {
+        // writers bytesl takes in a vec! and its length
+        let blob_length = v.len();
+        assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, blob_length));
+    }
+    unsafe fn write_usize(&self, mut u: &usize) {
+        write_length(self.w, u.to_be());
+    }
+}
+
+struct StructuredCloneReader {
+    r: *mut JSStructuredCloneReader,
+}
+
+impl StructuredCloneReader {
+    unsafe fn read_bytes(&self) -> usize {
+        // reads and return length
+        let type_length = read_length(self.r);
+        return type_length;
+    }
+    unsafe fn read_str(&self) -> String {
+        // return string from vec!
+        let type_str_length = read_length(self.r);
+        let mut type_str_buffer = vec![0u8; type_str_length];
+        return String::from_utf8_unchecked(type_str_buffer);
+    }
+    unsafe fn read_slice(&self) -> Vec<u8> {
+        // returns vec! with blob/str length
+        let blob_buffer = read_length(self.r);
+        let vect = vec![0u8; blob_buffer];
+        return vect;
+    }
+    unsafe fn read_assert(&self, v: &mut [u8], l: usize) {
```

Comment:
It is not clear what this method is supposed to do that is not handled by the previous methods.